frontend/components/CompactNavigationBar.js

```diff
@@ -0,0 +1,440 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Image from "next/image";
+import styles from "../styles/Navbar.module.css";
+
+// import menu icons
+import MenuIcon from "@material-ui/icons/Menu";
+import CloseIcon from "@material-ui/icons/Close";
+// import menuDropdownContainer icons
+import ChromeReaderModeOutlinedIcon from "@material-ui/icons/ChromeReaderModeOutlined";
+import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
+import PeopleOutlineIcon from "@material-ui/icons/PeopleOutline";
+import PhoneOutlinedIcon from "@material-ui/icons/PhoneOutlined";
+import InsertDriveFileOutlinedIcon from "@material-ui/icons/InsertDriveFileOutlined";
+import HeadsetMicOutlinedIcon from "@material-ui/icons/HeadsetMicOutlined";
+import LocalPrintshopOutlinedIcon from "@material-ui/icons/LocalPrintshopOutlined";
+import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
+import VideocamOutlinedIcon from "@material-ui/icons/VideocamOutlined";
+
+const CompactNavigationBar = () => {
+  const [clearNavBar, setClearNavBar] = useState(true);
+  const [hiddenNavBar, setHiddenNavBar] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [aboutUsDropdownOpen, setAboutUsDropdownOpen] = useState(false);
+  const [mediaDropdownOpen, setMediaDropdownOpen] = useState(false);
+  const [darkMode, setDarkMode] = useState(false);
+  const router = useRouter();
+  const pageLoading = false;
+
+  /* check whether device is in light or dark mode 
+     and change darkMode state accordingly */
+  useEffect(() => {
+    if (
+      window.matchMedia &&
+      window.matchMedia("(prefers-color-scheme: dark)").matches
+    ) {
+      setDarkMode(true);
+    }
+  }, []);
+
+  /* check whether user is at the bottom of the page when scrolling
+     and change hiddenNavBar state accordingly */
+  useEffect(() => {
+    window.addEventListener("scroll", checkBottomScreen);
+    return () => {
+      window.removeEventListener("scroll", checkBottomScreen);
+    };
+  }, []);
+
+  /* check whether user is at the top of the page when scrolling
+     and change clearNavBar state accordingly */
+  useEffect(() => {
+    checkTopScreen();
+    window.addEventListener("scroll", checkTopScreen);
+    return () => {
+      window.removeEventListener("scroll", checkTopScreen);
+    };
+  }, []);
+
+  /* check whether a page has been loaded when this component is rendered
+     and close menus if true */
+  useEffect(() => {
+    if (pageLoading) {
+      menuItemClick();
+    }
+  });
+
+  const checkBottomScreen = () => {
+    const checkBottom =
+      Math.ceil(window.innerHeight + window.scrollY) >=
+      document.documentElement.scrollHeight;
+    setHiddenNavBar(checkBottom);
+  };
+
+  const checkTopScreen = () => {
```

Comment:
Move the logic common in both `CompactNavigationBar` and `NavigationBar` to a new file and import them in. Seems like the logic is repeated (e.g. `checkTopScreen`, `changeAboutUsDropDown`)